src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationMiddleware.cs

```diff
@@ -43,27 +43,27 @@ public class OAuthAuthenticationMiddleware<TOptions, TNotifications> : Authentic
             // todo: review error handling
             if (string.IsNullOrWhiteSpace(Options.AuthenticationScheme))
             {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AuthenticationScheme)));
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options)));
```

Comment:
You are right @brentschmaltz, but we faced an issue with ```nameof``` while it returns the name of the last identifier in the property path. I know the latest change doesn't make sense from meaning point of you. I'm looking for a decent solution. Hope i hear from you guys :smile: